Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.swift

```diff
@@ -14,245 +14,243 @@
-                        .frame(maxWidth: .infinity, alignment: .center)
-                        .padding(8)
-                        .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
-                }
-                .onDisappear {
-                    Task { await locationDataSource.stop() }
+        MapView(
+            map: model.map,
+            graphicsOverlays: model.graphicsOverlays
+        )
+        .onSingleTapGesture { _, mapPoint in
+            Task {
+                do {
+                    try await model.addRouteGraphic(for: mapPoint)
+                } catch {
+                    self.error = error
                 }
-                Divider()
             }
-            .toolbar {
-                ToolbarItemGroup(placement: .bottomBar) {
-                    Button("Start") {
-                        isNavigating = true
-                        Task {
-                            do {
-                                try await startNavigation()
-                            } catch {
-                                print("Failed to start navigation.")
-                            }
-                        }
-                    }
-                    .disabled(isNavigating)
+        }
+        .locationDisplay(model.locationDisplay)
+        .task {
+            Task {
+                do {
+                    try await model.setup()
+                } catch {
+                    self.error = error
                 }
             }
         }
+        .overlay(alignment: .top) {
+            instructionText
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                toolbarButtons
+            }
+        }
+        .onAppear {
+            model.statusText = "Tap to place a start point."
+        }
+        .errorAlert(presentingError: $error)
     }
     
-    /// Creates a graphics overlay and adds a graphic (with solid yellow 3D tube symbol)
-    /// to represent the route.
-    private func makeRouteOverlay(routeResult: RouteResult, routeGraphic: Graphic) -> GraphicsOverlay {
-        let graphicsOverlay = GraphicsOverlay()
-        graphicsOverlay.sceneProperties.surfacePlacement = .absolute
-        let strokeSymbolLayer = SolidStrokeSymbolLayer(
-            width: 1.0,
-            color: .yellow,
-            lineStyle3D: .tube
-        )
-        let polylineSymbol = MultilayerPolylineSymbol(symbolLayers: [strokeSymbolLayer])
-        let polylineRenderer = SimpleRenderer(symbol: polylineSymbol)
-        graphicsOverlay.renderer = polylineRenderer
-        
-        if let originalPolyline = routeResult.routes.first?.geometry {
-            addingElevation(3, to: originalPolyline) { polyline in
-                routeGraphic.geometry = polyline
-                graphicsOverlay.addGraphic(routeGraphic)
+    /// The buttons in the bottom toolbar.
+    @ViewBuilder private var toolbarButtons: some View {
+        Spacer()
+        NavigationStack {
```

Comment:
This stack can be removed once #484 goes in but no need to hold up this PR for that.